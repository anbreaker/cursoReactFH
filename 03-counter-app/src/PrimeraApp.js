import React from 'react';

// Functional Component

const PrimeraApp = () => {
  // const saludo = {
  //   nombre: 'anbreaker',
  //   edad: 34,
  // };
  const saludo = 'Hola Mundo';
  return (
    <>
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <h1>{saludo}</h1>
      <p>Mi primera App React</p>
    </>
  );
};

export default PrimeraApp;
