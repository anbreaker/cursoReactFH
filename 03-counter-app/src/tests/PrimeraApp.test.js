import React from 'react';
import '@testing-library/jest-dom';
// import {render} from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import {shallow} from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {
  // Pruebas Jest Nativo
  // test('Debe mostrar el mensaje', () => {
  //   const saludo = 'Hola, soy Goku';
  //   const {getByText} = render(<PrimeraApp saludo={saludo} />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });
  //
  // Pruebas con Enzyme

  test('Debe mostrar <PrimeraApp /> correctamente ', () => {
    const saludo = 'Hola, soy Goku';
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });
});
