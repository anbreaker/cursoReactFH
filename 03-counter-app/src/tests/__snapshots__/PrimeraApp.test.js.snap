// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
